modules/core/03-connection/keeper/handshake.go

```diff
@@ -250,9 +250,9 @@ func (k Keeper) ConnOpenAck(
 	}
 
 	// Retrieve chainA's consensus state at consensusheight
-	expectedConsensusState, found := k.clientKeeper.GetSelfConsensusState(ctx, consensusHeight)
-	if !found {
-		return clienttypes.ErrSelfConsensusStateNotFound
+	expectedConsensusState, err := k.clientKeeper.GetSelfConsensusState(ctx, consensusHeight)
+	if err != nil {
+		return clienttypes.ErrSelfConsensusStateNotFound.Wrap(err.Error())
```

Comment:
```suggestion
	return sdkerrors.Wrapf(err, "self consensus state not found for height %s", consensusHeight.String())
```

Reply:
might be better to apply the changes yourself instead of committing the suggestion due to formatting and typing errors